src/main/view/game/UnitOverviewPanel.java

```diff
@@ -1,32 +1,210 @@
+	}
 
-	public UnitOverviewPanel(GameModelAdapter gameModelAdapter, AssetManager assets,
-			ViewEnum viewEnum) {
-		super(gameModelAdapter, assets, viewEnum);
-		setIsVisible(false);
+	private void setUpColumns() {
+		unitTable.getColumns().add(unitTypeColumn);
+		unitTable.getColumns().add(healthColumn);
+		unitTable.getColumns().add(attackColumn);
+		unitTable.getColumns().add(defenseColumn);
+		unitTable.getColumns().add(armorColumn);
+		unitTable.getColumns().add(upkeepColumn);
+		unitTable.getColumns().add(addToArmyColumn);
+
+		unitTypeColumn.setResizable(false);
+		attackColumn.setResizable(false);
+		healthColumn.setResizable(false);
+		defenseColumn.setResizable(false);
+		armorColumn.setResizable(false);
+		upkeepColumn.setResizable(false);
+		addToArmyColumn.setResizable(false);
+		
+		unitTypeColumn.setCellValueFactory(unitList -> unitList.getValue().unitTypeProp);
+		attackColumn.setCellValueFactory(unitList -> unitList.getValue().attackProp);
+		healthColumn.setCellValueFactory(unitList -> unitList.getValue().healthProp);
+		defenseColumn.setCellValueFactory(unitList -> unitList.getValue().defenseProp);
+		armorColumn.setCellValueFactory(unitList -> unitList.getValue().armorProp);
+		upkeepColumn.setCellValueFactory(unitList -> unitList.getValue().upkeepProp);
+		addToArmyColumn.setCellValueFactory(unitList -> unitList.getValue().armyProp);
 	}
 
 	public void draw(GraphicsContext g, Point screenDimensions, long currentPulse) {
-		drawPanelBox(g, screenDimensions);
-		Font oldFont = g.getFont();
-		g.setFont(getAssets().getFont(2));
-		g.fillText("Unit Overview", screenDimensions.x / 2 - 370, screenDimensions.y / 2 - 245);
-		g.setFont(getAssets().getFont(1));
+		scrollPane.toFront();
+		updatePositions(screenDimensions);
+		updateUnits();
 	}
 
-	public void hideGUIElements() {
+	private void updatePositions(Point screenDimensions) {
+		scrollPane.setMaxWidth(screenDimensions.x - 148);
+		unitTable.setPrefWidth(screenDimensions.x - 148);
+		scrollPane.setMaxHeight(488);
+		scrollPane.setTranslateX(74);
+		scrollPane.setTranslateY(50);
+		unitTypeColumn.setPrefWidth(unitTable.getWidth()/7);
+		healthColumn.setPrefWidth(unitTable.getWidth()/7 - 13);
+		attackColumn.setPrefWidth(unitTable.getWidth()/7 - 13);
+		defenseColumn.setPrefWidth(unitTable.getWidth()/7 - 13);
+		armorColumn.setPrefWidth(unitTable.getWidth()/7 - 13);
+		upkeepColumn.setPrefWidth(unitTable.getWidth()/7 - 13);
+		addToArmyColumn.setPrefWidth(unitTable.getWidth()/7 - 13);
+	}
+
+	private void updateUnits() {
+		ArrayList<Unit> units = getAdapter().getCurrentUnits();
+        unitTable.setPrefHeight(45 + (units.size() * 35));
+		unitList.clear();
+		for (Unit unit : units) {
+			String unitType = "";
+			switch (unit.getType()) {
+			case EXPLORER:
+				unitType = "Explorer";
+				break;
+			case COLONIST:
+				unitType = "Colonist";
+				break;
+			case MELEE:
+				unitType = "Melee";
+				break;
+			case RANGE:
+				unitType = "Ranged";
+				break;
+			default:
```

Comment:
so if its none of these types its still going to add a UnitItem just with a blank type